src/components/RouletteMain/RouletteMain.jsx

```diff
@@ -1,148 +1,234 @@
-import React, { Component } from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import Header from '../Header'
 import Footer from '../Footer'
 import ErrorBoundary from '../ErrorBoundary'
 import { ProcessGetRequest } from '../../utils/RestUtils'
 import AddEditMovie from '../AddEditMovie'
 import DeleteMovie from '../DeleteMovie'
-import SortListing from '../SortListing'
+import SortFilterListing from '../SortFilterListing'
+import MovieDetails from '../MovieDetails'
 import "regenerator-runtime"
 const MoviesList = React.lazy(() => import("../MoviesList"));
 
-class RouletteMain extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            movieList: [],
-            sortOption: "year", 
-            isError: false,
-            isAddModalVisible: false,
-            isEditModalVisible: false,
-            isDeleteModalVisible: false,
-            processEditMovieField: {},
-            processDeleteMovieField: {}
-        }
-    }
+export default function RouletteMain(props) {
+    const [isError, setError] = useState(false);
+    const [movieList, setMovieList] = useState([]);
```

Comment:
To use custom hook you should invoke it: ```const {response, error} = useFetch();```. After that you should update your state based on the value from the custom hook:
```
 useEffect(() => {
        setMovieList(response || []);
    }, [response]);
```